### PR TITLE
Add support for cticks

### PR DESCRIPTION
**Pull request by ahuang11**
Closes https://github.com/holoviz/holoviews/issues/5723
```python
import numpy as np
import holoviews as hv

hv.extension("bokeh")

img = hv.Image(np.array([[0, 1], [2, 3]])).opts(
    cticks=[(1, "A"), (2, "B")], colorbar=True
)
img
```

image

Didn't implement for matplotlib because it already has `cbar_ticks`

### CODE REVIEW

File: holoviews/plotting/bokeh/element.py [hunk 1]
```diff
@@ -1051,28 +1051,8 @@ def _axis_properties(self, axis, key, plot, dimension=None,
...
-                else:
-                    ticks, labels = ticker, None
-                if ticks and util.isdatetime(ticks[0]):
-                    ticks = [util.dt_to_int(tick, 'ms') for tick in ticks]
-                axis_props['ticker'] = FixedTicker(ticks=ticks)
-                if labels is not None:
-                    axis_props['major_label_overrides'] = dict(zip(ticks, labels))
+            if not (self._subcoord_overlaid and axis == 'y'):
+                axis_props.update(get_ticker_axis_props(ticker))
             elif self._subcoord_overlaid and axis == 'y':
```
Comment (hoxbro) [inline holoviews/plotting/bokeh/element.py:1056]:
```suggestion
            else:
```

Or move the if-part down to the else.

File: holoviews/plotting/bokeh/element.py [hunk 2]
```diff
@@ -2470,6 +2450,11 @@ class ColorbarPlot(ElementPlot):
         #FFFFFFFF or a length 3 or length 4 tuple specifying values in
         the range 0-1 or a named HTML color.""")
 
+    cticks = param.Parameter(default=None, doc="""
```
Comment (hoxbro) [inline holoviews/plotting/bokeh/element.py:2453]: Can we use `param.ClassSelector` here?

Even if it is verbose, it makes it easier to understand what is permitted.  You are also welcome to update `xticks` and `yticks` if they are also parameters.

File: holoviews/plotting/bokeh/util.py [hunk 3]
```diff
@@ -1167,3 +1173,30 @@ def dtype_fix_hook(plot, element):
                 for k, v in data.items():
                     if hasattr(v, "dtype") and v.dtype.kind == "U":
                         data[k] = v.tolist()
+
+
+def get_ticker_axis_props(ticker):
```
Comment (hoxbro) [inline holoviews/plotting/bokeh/util.py:1178]: I haven't checked, but this is just an extraction, correct?

File: holoviews/plotting/bokeh/element.py (hunk 2 above)
Comment (ahuang11) [reply holoviews/plotting/bokeh/element.py]: :| Very verbose indeed.

File: holoviews/plotting/plot.py [hunk 4]
```diff
@@ -1132,15 +1132,13 @@ class GenericElementPlot(DimensionedPlot):
     yrotation = param.Integer(default=None, bounds=(0, 360), doc="""
         Rotation angle of the yticks.""")
 
-    xticks = param.Parameter(default=None, doc="""
+    xticks = param.ClassSelector(class_=(int, list, tuple, np.ndarray), default=None, doc="""
```
Comment (ahuang11) [inline holoviews/plotting/plot.py:1135]: Not specific to Bokeh.

**Pull request merged**